python/llm/example/CPU/HF-Transformers-AutoModels/Model/falcon/falcon-7b-instruct/modelling_RW.py

```diff
@@ -294,7 +295,13 @@ def forward(
             _, seq_len_past, _ = layer_past[0].shape
 
             seq_len = seq_len + seq_len_past
-            position_ids = torch.tensor([range(seq_len_past, seq_len)], dtype=torch.long).to(query_layer.device)
+            if (seq_len_past, seq_len) in self.p_dict:
+                #print("get cache, " + str(seq_len_past) + "  " + str(seq_len))
+                position_ids = self.p_dict[(seq_len_past, seq_len)]
+            else:
+                #print("create new" + str(seq_len_past) + "  " + str(seq_len))
+                position_ids = torch.tensor([range(seq_len_past, seq_len)], dtype=torch.long).to(query_layer.device)
+                self.p_dict[(seq_len_past, seq_len)] = position_ids
```

Comment:
do we need to store every seq_len?

Reply:
Yes, store every seq_len is better than `new every time` and `cache one, and assign value`.
The best option should be add a new kernel function to accept seq_len directly.